tests/ldap/test_ldap.py

```diff
@@ -547,5 +522,65 @@ def _prepare():
 
     # make sure account 2 was deleted
     with pytest.raises(NoResultFound):
-        ra2 = RemoteAccount.query.filter(
+        RemoteAccount.query.filter(
             RemoteAccount.user_id == user_to_delete_id2).one()
+
+
+def test_send_email_on_user_deletion_error(app_with_notifs, mocker):
+    """Test that an email is sent when the user deletion failed."""
+    # mock the users exit in LDAP
+
+    def mock_get_patron(patron_id):
+        from cds_ils.patrons.api import Patron
+
+        class FakePatron(Patron):
+            def __init__(self, id, revision_id=None):
+                self.id = id
+                self.revision_id = 1
+                self.name = f"Patron{id}"
+                self.email = f"patron{id}@cern.ch"
+        return FakePatron(patron_id)
+
+    mock_map = Mock()
+    mock_map.get.return_value = True
+    mocker.patch("cds_ils.ldap.api.get_ldap_users",
+                 return_value=(2, {}, {}))
+    mocker.patch("invenio_app_ils.patrons.anonymization.current_app_ils.patron_cls.get_patron", mock_get_patron)  # noqa
+    mocker.patch("invenio_app_ils.patrons.anonymization.current_app_ils.patron_indexer.delete")  # noqa
+    # mock RemoteAccounts
+    mock1 = Mock()
+    mock1.user_id = 1  # patron 1
+    mock2 = Mock()
+    mock2.user_id = 2  # patron 2
+    mocker.patch("cds_ils.ldap.api.remap_invenio_users",
+                 return_value=[mock1, mock2])
+    # mock that the any `InvenioUser` exist
+    mocker.patch("cds_ils.ldap.api.InvenioUser")
```

Comment:
is this working in the end?

Reply:
yes I changed the mock from where it was implemented to where it was used, the same with the mock below